registry/remote/credentials/memory_store.go

```diff
@@ -17,23 +17,50 @@ package credentials
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"sync"
 
 	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/credentials/internal/config"
 )
 
-// memoryStore is a store that keeps credentials in memory.
-type memoryStore struct {
+// MemoryStore is a store that keeps credentials in memory.
+type MemoryStore struct {
 	store sync.Map
 }
 
 // NewMemoryStore creates a new in-memory credentials store.
 func NewMemoryStore() Store {
-	return &memoryStore{}
+	return &MemoryStore{}
+}
+
+// NewMemoryStoreFromConfig creates a new in-memory credentials store from the given configuration.
+//
+// Reference: https://docs.docker.com/engine/reference/commandline/cli/#docker-cli-configuration-file-configjson-properties
+func NewMemoryStoreFromConfig(c []byte) (*MemoryStore, error) {
+	cfg := struct {
+		Auths map[string]config.AuthConfig `json:"auths"`
+	}{}
+	if err := json.Unmarshal(c, &cfg); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal auth field: %w: %v", config.ErrInvalidConfigFormat, err)
+	}
+
+	s := &MemoryStore{}
+	for addr, auth := range cfg.Auths {
+		// Normalize the auth key to hostname.
+		hostname := config.ToHostname(addr)
+		cred, err := auth.Credential()
+		if err != nil {
+			return nil, err
+		}
+		s.store.Store(hostname, cred)
+	}
+	return s, nil
```

Comment:
It changes the behavior. We should also allow exact match.

Reply:
Consider
```suggestion
	s := &MemoryStore{}
	for addr, auth := range cfg.Auths {
		cred, err := auth.Credential()
		if err != nil {
			return nil, err
		}
		s.store.Store(addr, cred)
		// normalize the auth key to hostname.
		_, _ = s.store.LoadOrStore(hostname, cred)
	}
	return s, nil
```

Reply:
I use `LoadOrStore` to prevent overwriting.

Reply:
/cc @Wwwsylvia for comments

Reply:
I realized that we can just normalize the host name, as the auth module only looks for the host name. Actually my memory was wrong, the original implementation did not do exact match.

Reply:
Oh, OK. Let me close this comment.